lib/publiccloud/utils.pm

```diff
@@ -885,4 +887,104 @@ sub upload_asset_on_remote {
     $instance->ssh_assert_script_run($mv_cmd);
 }
 
+
+=head2 zypper_remote_call
+
+    zypper_remote_call($command [, exitcode => $exitcode] [, timeout => $timeout];
+
+Function wrapping zypper or transactional-update command for remote execution via ssh; not for tunneling.
+Implements lib/utils::zypper_call, dedicated to publiccloud instances, but the input command is totally up to the user, 
+simply expecting zypper or transactional-update commands present, being some zypper commands allowed in transactionals.
+Unlike 'zypper_call', here the zypper_log_packages preparation is skipped.
+
+Usage example:
+    $instance->zypper_remote_call("LANG=C sudo zypper -n up", exitcode => [0,102,103], timeout => 300);
+
+=cut
+
+sub zypper_remote_call {
+    my $instance = shift;
+    my %args = testapi::compat_args({cmd => undef}, ['cmd'], @_);
+    my $command = $args{cmd};
+    # check command line:
+    my $type = $command =~ (/((^|\b)zypper\b)(?![-\/])/) ? 1 :
+      ($command =~ /((^|\b)transactional-update\b)(?![-\/])/) ? 2 : 0;
+    die("ERROR: unexpected command: " . $command) unless ($type);
+    #
+    $args{rc_only} = 1;
+    $args{timeout} //= 700;
+    my $exit = $args{exitcode} || [0];
+    my $retry = $args{retry} // 2;
+    my $delay = $args{delay} // 5;
+    my $log = "/var/log/zypper.log";
+    my $ret;
+    #
+    delete $args{exitcode};
+    delete $args{retry};
+    delete $args{delay};
+    # retry loop
+    for (1 .. $retry) {
+        # pause on next
+        sleep($delay) if (defined($ret));
+        $ret = $instance->run_ssh_command(%args);
+        die "FAILED: timeout after " . $args{timeout} . " sec." unless defined($ret);
```

Comment:
the fact that run_ssh_command returns non-zero value not necessarily means that command failed on timeout . non-zero value will be returned also in case something failed in zypper call